assets/src/components/activities/oli_embedded/OliEmbeddedAuthoring.tsx

```diff
@@ -0,0 +1,250 @@
+import { ActivityXmlEditor } from 'components/common/ActivityXmlEditor';
+import ModalSelection from 'components/modal/ModalSelection';
+const store = configureStore();
+
+const Embedded = (props: AuthoringElementProps<OliEmbeddedModelSchema>) => {
+  const { dispatch, model, onRequestMedia } = useAuthoringElementContext<OliEmbeddedModelSchema>();
+
+  const { projectSlug } = props;
+
+  function select(_projectSlug: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      const request = {
+        type: 'MediaItemRequest',
+      } as MediaItemRequest;
+      if (props.onRequestMedia) {
+        props.onRequestMedia(request).then((r) => {
+          if (r === false) {
+            reject('error');
+          } else {
+            resolve(r as string);
+          }
+        });
+      }
+    });
+  }
+
+  const addFile = (e: any) => {
+    select(projectSlug).then((url: string) => {
+      dispatch(OliEmbeddedActions.addResourceURL(url));
+    });
+  };
+
+  const onFileUpload = (files: FileList) => {
+    // get a list of the files to upload
+    const fileList: File[] = [];
+    for (let i = 0; i < files.length; i = i + 1) {
+      const file = files[i];
+      fileList.push(renameFile(file, 'webcontent/' + model.resourceBase + '/' + file.name));
+    }
+
+    uploadFiles(projectSlug, fileList)
+      .then((result: any) => {
+        result.forEach((i: any) => {
+          dispatch(OliEmbeddedActions.addResourceURL(i.url));
+        });
+      })
+      .catch((reason: any) => {
+        console.log(JSON.stringify(reason.message));
+        display(errorModal(reason.message), '#upload_error');
+      });
+  };
+
+  const renameFile = (originalFile: any, newName: string) => {
+    return new File([originalFile], newName, {
+      type: originalFile.type,
+      lastModified: originalFile.lastModified,
+    });
+  };
+
+  const onUploadClick = (id: string) => {
+    (window as any).$('#' + id).trigger('click');
+  };
+
+  const handleScoringChange = (partId: string, key: string) => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const scoring: ScoringStrategy = ScoringStrategy[key];
+    dispatch(OliEmbeddedActions.updatePartScoringStrategy(partId, scoring));
+  };
+
+  const removePart = (partId: string) => {
+    dispatch(OliEmbeddedActions.removePart(partId));
+  };
+
+  const addNewPart = () => {
+    dispatch(OliEmbeddedActions.addNewPart());
+  };
+
+  const id = guid();
+
+  const errorModal = (error: string) => {
+    const footer = (
+      <>
+        <button
+          type="button"
+          className="btn btn-primary"
+          onClick={() => {
+            dismiss('error_message');
+          }}
+        >
+          Ok
+        </button>
+      </>
+    );
+
+    return (
+      <ModalSelection
+        title="File Upload"
+        footer={footer}
+        onCancel={() => {
+          dismiss('error_message');
```

Comment:
in the implementation of dismiss these are using querySelector, so I think you need the `#` here. OR maybe some logic inside the dismiss method to add it if it's missing

Reply:
Hardcoded value was wrong. Replaced with id parameter